### PR TITLE
Fixes emag not using charges

### LINKED ISSUE #7472: EMAGs opens all doors on station without using charges
### Environment
- Game Version: Staging 21092617
- OS Version: Arch Linux

### Detailed description of bug

The EMAG when hold on the hand opens all doors and locks them open like the bolt toggle ID, and doesnt use charges when doing that.

https://user-images.githubusercontent.com/80334192/134824978-e59e0fdb-acd3-4e7e-b4e6-28f0cb9b67e8.mp4

### Steps to reproduce

1. spawn an EMAG and run to door to open it
2. to make sure its the emag take off any IDs from your pockets

### PR DESCRIPTION
**Pull request by MaxIsJoe**
fixes #7472 

CL: [Fix] Fixed emag not using charges

### CODE REVIEW

**Review by Bod9001 — DISMISSED** [review]

File: UnityProject/Assets/Scripts/Objects/Doors/Modules/EmagInteractionModule.cs [hunk 1]
```diff
@@ -39,15 +39,15 @@ public override ModuleSignal ClosedInteraction(HandApply interaction, HashSet<Do
 						foreach (var item in ItemStorage.GetNamedItemSlots(NamedSlot.id))
 						{
 							var ID = item.ItemAttributes;
+							Emag emagInIdSlot = item.Item?.OrNull().gameObject.GetComponent<Emag>()?.OrNull();
 
-							if (ID != null)
+							if (ID != null) {continue;}
+							if (emagInIdSlot != null) {continue;}
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Objects/Doors/Modules/EmagInteractionModule.cs]: excess brackets

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Objects/Doors/Modules/EmagInteractionModule.cs]: fixed

**Review by Bod9001 — DISMISSED** [review]

File: UnityProject/Assets/Scripts/Objects/Doors/Modules/EmagInteractionModule.cs [hunk 2]
```diff
@@ -39,15 +39,15 @@ public override ModuleSignal ClosedInteraction(HandApply interaction, HashSet<Do
 						foreach (var item in ItemStorage.GetNamedItemSlots(NamedSlot.id))
 						{
 							var ID = item.ItemAttributes;
+							Emag emagInIdSlot = item.Item?.OrNull().gameObject.GetComponent<Emag>()?.OrNull();
 
-							if (ID != null)
+							if (ID != null) continue;
+							if (emagInIdSlot != null) continue;
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Objects/Doors/Modules/EmagInteractionModule.cs]:
```suggestion
							if (ID == null) continue;
							if (emagInIdSlot == null) continue;
```

Just noticed that

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Objects/Doors/Modules/EmagInteractionModule.cs]: why would we continue if it's null? do you want to throw NREs?

Comment (ThatDan123) [reply UnityProject/Assets/Scripts/Objects/Doors/Modules/EmagInteractionModule.cs]: You continue to avoid the NRE

Comment (ThatDan123) [reply UnityProject/Assets/Scripts/Objects/Doors/Modules/EmagInteractionModule.cs]: also `ID` isnt used anywhere so can be removed

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Objects/Doors/Modules/EmagInteractionModule.cs]: done

**Comment by ThatDan123** [comment]
Looked through the whole script and from line 66 is just a duplication of the stuff you just needed to change, can be split off into into own function and called from both interactions with a module signal return. 

Sorry that should be the last change needed.

**Comment by MaxIsJoe** [comment]
Done.

**Review by ThatDan123 — APPROVED** [review]

**Review by Bod9001 — APPROVED** [review]

**Pull request merged**